src/lib/y2partitioner/widgets/encrypt_password.rb

```diff
@@ -93,6 +73,94 @@ def pw1
       def pw2
         Yast::UI.QueryWidget(Id(:pw2), :Value)
       end
+
+      # @return [Boolean]
+      def random_password_allowed?
+        @random_password_allowed ||= @controller.filesystem_type.to_sym == :swap
+      end
+
+      # @return [Boolean]
+      def random_password?
+        return false unless random_password_allowed?
+
+        Yast::UI::QueryWidget(Id(:pwd_type), :Value) == :random_pwd
+      end
+
+      def form_content
+        if random_password_allowed?
+          extended_content
+        else
+          password_fields
+        end
+      end
+
+      def extended_content
+        VBox(
+          RadioButtonGroup(
+            Id(:pwd_type),
+            VBox(
+              Left(RadioButton(Id(:manual_pwd), _("Ask for password on boot to mount the swap"), true)),
+              password_fields,
```

Comment:
well, I expect here can be more playing. How it does look in UI as those password fields is realted only to this radio button? I think that also enable/disable when you check/uncheck this radio button makes sense for me. And as this started to be more complicated I suggest to use here another object for this password fields that will be included in this widget as it starting to be more complicated and deserve some split.

I also expect that it more naturally allow better and more readable code in other parts which now heavilty mixture this password fields with other radio button selection.

Reply:
I agree. Password fields should be enabled/disabled according to the selected option. I forgot to add that question to the PR description since I did not investigated/discovered yet how you do that in YaST UI. Links are welcome :)

Regarding complexity, I feel the same but I need more time to figure out how to improve the code. That's the reason why I opened the PR as WIP, for early feedback / help.

So, let me a couple of days to improve it :)